### PR TITLE
When DB is locked, it constantly polls for lock based on DATABASE_POLL_INTERVAL

### PR DESCRIPTION
**Pull request by jleeh**
I struggled to create a unit test for this, I got to a point where I modified cltest to include a function where it created a new application with the same root directory. Although on testing, it obviously just locked the test up because it was waiting on a DB lock.

For issue: #330 
[[2129823](https://www.pivotaltracker.com/n/projects/2129823)]

### CODE REVIEW

**Comment by se3000** [comment]
Yeah, sounds like a hard thing to test. Will check it out and get back to you.

File: store/config.go [hunk 1]
```diff
@@ -124,11 +129,22 @@ func levelParser(str string) (interface{}, error) {
...
+}
+
 // LogLevel determines the verbosity of the events to be logged.
 type LogLevel struct {
 	zapcore.Level
 }
 
+// Duration returns a time duration with the supported
+// units of "ns", "us", "ms", "s", "m", "h".
+type Duration struct {
```
[the remark this answers is not in the document]
Comment (jleeh) [reply store/config.go]: The reflect.TypeOf method is expecting a composite literal, whereas time.Duration is an int64.

**Comment by jleeh** [comment]
Increased the log level of the sleep msg also, db outputs an error also if it's locked, thought it'd be better to show even when not in debug:
```
2018-06-07T22:28:40Z [INFO]  BoltDB is locked, sleepingsleepDuration1s          store/store.go:121               
bolt.Close(): funlock error: The segment is already unlocked.
2018-06-07T22:28:42Z [INFO]  BoltDB is locked, sleepingsleepDuration1s          store/store.go:121               
bolt.Close(): funlock error: The segment is already unlocked.
```

File: store/store.go [hunk 2]
```diff
@@ -109,3 +110,19 @@ func (Clock) Now() time.Time {
...
 	return time.After(d)
 }
+
+func initializeORM(config Config) *models.ORM {
+	var orm *models.ORM
+	var err error
+	sleeper := utils.NewConstantSleeper(config.DatabasePollInterval.Duration)
+	for {
+		orm, err = models.NewORM(path.Join(config.RootDir, "db.bolt"))
+		if err != nil && err.Error() == "timeout" {
```
Comment (se3000) [inline store/store.go]: Last small thing: We try to match [the error](https://github.com/coreos/bbolt/blob/master/errors.go#L28) if possible, it's more exact than string comparison.

Something like `if err != nil && err == bolt.ErrTimeout {...` You might have to import bbolt as bolt: `bolt "github.com/coreos/bbolt"`

**Comment by jleeh** [comment]
Just got this updated @se3000.

**Comment by se3000** [comment]
Thanks @jleeh, makes updates much simpler.

**Pull request merged**